src/explore-education-statistics-admin/src/pages/publication/components/__tests__/LegacyReleasesTable.test.tsx

```diff
@@ -0,0 +1,283 @@
+      );
+      userEvent.click(
+        screen.getAllByRole('button', { name: 'Edit release' })[0],
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Warning'));
+      });
+
+      userEvent.click(screen.getByRole('button', { name: 'OK' }));
+
+      await waitFor(() => {
+        expect(history.location.pathname).toBe(
+          `/publication/${testPublication.id}/legacy-releases/${testPublication.legacyReleases[0].id}/edit`,
+        );
+      });
+    });
+  });
+
+  describe('deleting', () => {
+    test('shows a warning modal when the delete release button is clicked', async () => {
+      render(<LegacyReleasesTable publication={testPublication} />);
+      userEvent.click(
+        screen.getAllByRole('button', { name: 'Delete release' })[0],
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Delete legacy release'));
+      });
+      expect(
+        screen.getByText(
+          'Are you sure you want to delete this legacy release?',
+        ),
+      );
+      expect(
+        screen.getByText(
+          'All changes made to legacy releases appear immediately on the public website.',
+        ),
+      );
+      expect(screen.getByRole('button', { name: 'Cancel' }));
+      expect(screen.getByRole('button', { name: 'Confirm' }));
+    });
+
+    test('sends the delete request and updates the releases table when confirm is clicked', async () => {
+      render(<LegacyReleasesTable publication={testPublication} />);
+      userEvent.click(
+        screen.getAllByRole('button', { name: 'Delete release' })[0],
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Delete legacy release'));
+      });
+      userEvent.click(screen.getByRole('button', { name: 'Confirm' }));
+
+      expect(legacyReleaseService.deleteLegacyRelease).toHaveBeenCalledWith(
+        testPublication.legacyReleases[0].id,
+      );
+
+      await waitFor(() => {
+        const table = screen.getByRole('table');
+        const rows = within(table).getAllByRole('row');
+        expect(rows).toHaveLength(3);
+      });
+    });
+  });
+
+  describe('creating', () => {
+    test('shows a warning modal when the create legacy release button is clicked', async () => {
+      const history = createMemoryHistory();
+      render(
+        <Router history={history}>
+          <LegacyReleasesTable publication={testPublication} />
+        </Router>,
+      );
+      userEvent.click(
+        screen.getByRole('button', { name: 'Create legacy release' }),
+      );
+      await waitFor(() => {
+        expect(screen.getByText('Warning'));
+      });
+      expect(
+        screen.getByText(
+          'All changes made to legacy releases appear immediately on the public website.',
+        ),
+      );
+      expect(screen.getByRole('button', { name: 'Cancel' }));
+      expect(screen.getByRole('button', { name: 'OK' }));
+    });
+    test('goes to the create page when OK is clicked', async () => {
```

Comment:
Should have line breaks between test cases

Reply:
Done